internal/reconciler/implementation.go

```diff
@@ -115,41 +131,9 @@ func (r *Implementation) Reconcile(ctx context.Context, req reconcile.Request) (
 		logger.Error(validationError, webhookValidationErrorLogMsg)
```

Comment:
What do you think about moving this error handling to line 105? I think it makes sense to handle the error right after receiving it, and if we end up returning because of the context being canceled the error won't be handled at all.

Reply:
makes sense